pyteal/ast/subroutine.py

```diff
@@ -136,28 +151,55 @@ def has_return(self):
 
 
 class SubroutineCall(Expr):
-    def __init__(self, subroutine: SubroutineDefinition, args: List[Expr]) -> None:
+    def __init__(
+        self, subroutine: SubroutineDefinition, args: List[Union[Expr, ScratchVar]]
+    ) -> None:
         super().__init__()
         self.subroutine = subroutine
         self.args = args
 
         for i, arg in enumerate(args):
-            if arg.type_of() == TealType.none:
+            arg_type = None
+
+            if not isinstance(arg, (Expr, ScratchVar)):
```

Comment:
I don't see any checks to make sure the arguments match the specific subroutine definition. Meaning if the subroutine has args `(ScratchVar, ScratchVar)`, you could currently pass in `(Expr, Expr)` without any errors.

Reply:
👍